File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        ans, l, total = inf, 0, 0
        for r in range(len(nums)):
            total += nums[r]
            while total >= target:
                ans = min(ans, r - l + 1)
                total -= nums[l]
                l += 1
        return 0 if ans == inf else ans

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int minLen = Integer.MAX_VALUE;
        int total = 0, left = 0, right = 0;
        while(right < nums.length){
            total += nums[right];
            right++;
            while(left <= right && total >= target){
                minLen = Math.min(minLen, right - left);
                total -= nums[left];
                left += 1;
            }
        }
        return minLen == Integer.MAX_VALUE ? 0 : minLen;
    }
}